src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -126,7 +125,12 @@ export default function Swap({
   } = useDerivedSwapInfo()
 
   // detects trade load
-  const quoteLoading = useIsQuoteLoading()
+  const [quote, quoteLoading] = useGetQuoteAndStatus({ token: INPUT.currencyId, chainId })
+
+  // major quote load indication (param change, token change etc)
+  // opposite could be just a simple timeout check
+  const isQuoteHardLoading = quoteLoading && !quote?.price.amount
+  const isQuoteSoftLoading = quoteLoading && !isQuoteHardLoading
```

Comment:
1. I like how you detect the hard/soft loading state!
2. I like that you give it a name as a const, so you can use it
3. nit: Although I understand the concept of hard/soft, I would prefer to model it with concepts closer to the reality. Hard/Soft can be many things. I would suggest `isGettingNewQuote` or `isRefreshingQuote`